src/view/report.rs

```diff
@@ -58,9 +100,12 @@ impl<'a> fmt::Display for Report<'a> {
     }
 }
 
-pub fn show_activities<'a>(activities: &'a [&'a activity::Activity]) {
+pub fn show_activities<'a>(activities: &'a [&'a activity::Activity], format: Format) {
     let report = Report::new(activities);
-    println!("\n{}", report);
+    match format {
+        Format::SHELL => println!("{}", report),
+        Format::JSON => println!("{}", serde_json::to_string(&report).unwrap()),
```

Comment:
```suggestion
        Format::JSON => println!("{}", serde_json::to_string(&report).unwrap_or_else(|err| {
        println!("Could not parse report.\nSerde reported:\n{}", err);
        std::process::exit(1);
        })),
```

I would suggest using something like the above to handle the error case.
That prevents a `panic` and gives more information as to what happened. (Disclaimer: I have not yet tested the code above.)